### PR TITLE
refactor(*): De-angular pipelineConfigProvider; rename to PipelineRegistry

### PR DESCRIPTION
**Pull request by jrsquared**

### CODE REVIEW

File: app/scripts/modules/core/src/pipeline/config/PipelineRegistry.ts [hunk 1]
```diff
@@ -172,7 +171,7 @@ export class PipelineConfigProvider {
     if (candidates.length) {
       baseKey = candidates[0].provides;
     }
-    return this.getStageTypes().filter(stageType => {
+    return stageTypes.filter(stageType => {
```
Comment (anotherchrisberry) [inline app/scripts/modules/core/src/pipeline/config/PipelineRegistry.ts]: `getStageType` returns a deep copy of the stage types. Not sure if that's really important (let's find out?).

File: app/scripts/modules/core/src/pipeline/config/PipelineRegistry.ts [hunk 2]
```diff
@@ -242,16 +240,9 @@ export class PipelineConfigProvider {
     return this.getManualExecutionComponent(this.getTriggerConfig(triggerType)) !== null;
   }
 
-  // TODO: How do i make sure this works?
```
Comment (anotherchrisberry) [inline app/scripts/modules/core/src/pipeline/config/PipelineRegistry.ts]: `i` should be capitalized, i.e. `I`

**Comment by anotherchrisberry** [comment]
LGTM

File: app/scripts/modules/core/src/pipeline/config/PipelineRegistry.ts (hunk 2 above)
Comment (jrsquared) [reply app/scripts/modules/core/src/pipeline/config/PipelineRegistry.ts]: That's the reason I deleted it.

Comment (anotherchrisberry) [reply app/scripts/modules/core/src/pipeline/config/PipelineRegistry.ts]: What if you added it back?

Comment (jrsquared) [reply app/scripts/modules/core/src/pipeline/config/PipelineRegistry.ts]: Then thankfully I have this comment to help remind me.

**Pull request merged**